### PR TITLE
Implement hardware accelerated CRC32 and SHA1, using them if possible

### PR DESCRIPTION
**Pull request by CasualPokePlayer**
Resolves https://github.com/TASEmulators/BizHawk/issues/3347

CRC32's generic function is also replaced with zlib's as it is much more performant than our implementation.
Full hash of a ~731MB disc took only ~369 ms with this, and the generic CRC32 isn't so far behind at ~659 ms.
We probably should consider just always doing a full disc hash with this change, as it no longer takes around 4.3 seconds to do a full disc hash.

Also if someone's computer supports SHA instructions I'd like them to test this as I'm unable to do so with my PC.

- [x] I, the committer, have read the [licensing terms for contributors](https://github.com/TASEmulators/BizHawk/blob/master/contributing.md#copyrights-and-licensing) (last updated 2022-07-15) and am compliant

### CODE REVIEW

File: ExternalProjects/LibBizHash/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+LibBizHash is the unmanaged side for BizHawk's hashing.
+
+CRC32 code is taken from [zlib-ng](https://github.com/zlib-ng/zlib-ng) with massive slashing of code and various tweaks. This code is licensed under the zlib license.
+SHA1 is code is taken from [SHA-Intrinsics](https://github.com/noloader/SHA-Intrinsics) with some tweaks. This code is under the public domain.
+All modifications by the BizHawk team are licensed under MIT.
```
Comment (YoshiRulz) [inline ExternalProjects/LibBizHash/README.md:5]: nitpick: "modifications" are not licensed, the modified copy of the source is. More importantly this should include build instructions.

File: src/BizHawk.Common/checksums/CRC32.cs [hunk 2]
```diff
@@ -66,69 +83,41 @@ private static uint gf2_matrix_times(ReadOnlySpan<uint> mat, uint vec)
...
-				// if no more bits set, then done
-			} while (len != 0U);
-
-			// finally, combine and return
-			_current ^= crc;
+				len >>= 1;
+			} while (len != 0U); // if no more bits set, then done
+
+			// finally, combine and return
+			_current ^= crc;
```
Comment (YoshiRulz) [inline src/BizHawk.Common/checksums/CRC32.cs:120]: no whitespace-only changes please

Comment (CasualPokePlayer) [reply src/BizHawk.Common/checksums/CRC32.cs:120]: I think most of that was Visual Studio being a butt here with newlines, not sure how I fix this.

**Review by Morilli — APPROVED** [review]
Changes look good from my side.
According to the SHA and CRC tests, everything works fine.
I can't really test the performance improvement of the fast SHA implementation given I don't have a capable system, but I'm certain there's a ~4x speedup here for people that do.

Feel free to check my commit again; I tested it, but another look can't hurt.

**Pull request merged**